src/bin/lychee/options.rs

```diff
@@ -245,6 +245,15 @@ pub struct Config {
     #[structopt(short, long, default_value = "string")]
     #[serde(default)]
     pub format: Format,
+
+    /// Enable recursion (make sub-requests for detected links)
+    #[structopt(short, long)]
+    #[serde(default)]
+    pub recursive: bool,
+
+    /// Stop link checking beyond this maximum recursion depth. (Recommended for large inputs.)
+    #[structopt(long)]
+    pub depth: Option<usize>,
```

Comment:
I'd consider allowing this to be signed (e.g. `isize`), so that users could explicitly express the need for infinite recursion with numbers less than zero (e.g. `-1`).